src/datachain/lib/meta_formats.py

```diff
@@ -54,10 +54,10 @@ def read_schema(source_file, data_type="csv", expr=None, model_name=None):
     try:
         with source_file.open() as fd:  # CSV can be larger than memory
             if data_type == "csv":
-                data_string += fd.readline().decode("utf-8", "ignore").replace("\r", "")
```

Comment:
seems a bit unrelated? just to make sure?

Reply:
I had some issue with examples that are using this one after changes, and solution was to use `type="text"` when doing `.from_storage()` in `read_meta` function, with which we avoided need to decode to string as it was reading automatically as `TextFile` instead of `File` which is binary by default. I think it's more elegant this way anyway.

Reply:
> I had some issue with examples that are using this one after changes

do you remember the context by chance? any explanation to that?

Reply:
Looks like example that was failing to me (`/examples/get_started/json-csv-reader.py`) works even without these changes so they are not really required in this PR. Maybe issue occurred to me during development that was not related, idk ... anyway, I still think these changes are good as we don't need to explicitly decode strings if we read storage as `text` and I would leave them.